src/ctapipe/image/extractor.py

```diff
@@ -1295,6 +1300,23 @@ def __call__(
         )
 
 
+class VarianceExtractor(ImageExtractor):
+
+    """
+    Calculates the variance of each sample waveform
+
+    """
+
+    def __call__(self, waveforms, tel_id, trigger_time) -> DL1PedestalVarianceContainer:
```

Comment:
Is this really an image extractor? It doesn't follow the abstract API of `ImageExtractor`, it returns a different data type and it's arguments are also different.

For it to *be* an image extractor, it must be usable if configured through the configuration system.

This fails for example:

```
❯ ctapipe-process -i dataset://gamma_prod5.simtel.zst -o /tmp/test.dl1.h5 --CameraCalibrator.image_extractor_type=VarianceExtractor

2024-07-10 15:28:47,988 WARNING [ctapipe.core.telescope_component] (telescope_component.attach_subarray): TelescopeParameter type argument 'SST_1M_*' did not match any known telescope types
2024-07-10 15:28:49,210 ERROR [ctapipe.ctapipe-process] (tool.run): Caught unexpected exception: VarianceExtractor.__call__() got an unexpected keyword argument 'selected_gain_channel'
Traceback (most recent call last):
  File "/home/mnoethe/Uni/CTA/ctapipe/src/ctapipe/core/tool.py", line 426, in run
    self.start()
  File "/home/mnoethe/Uni/CTA/ctapipe/src/ctapipe/tools/process.py", line 311, in start
    self.calibrate(event)
  File "/home/mnoethe/Uni/CTA/ctapipe/src/ctapipe/calib/camera/calibrator.py", line 329, in __call__
    self._calibrate_dl1(event, tel_id)
  File "/home/mnoethe/Uni/CTA/ctapipe/src/ctapipe/calib/camera/calibrator.py", line 277, in _calibrate_dl1
    dl1 = extractor(
          ^^^^^^^^^^
TypeError: VarianceExtractor.__call__() got an unexpected keyword argument 'selected_gain_channel'
```